### PR TITLE
feat(datasync): add global conflict configuration

### LINKED ISSUE #1902: Add Global conflict config for datasync
Following up from https://github.com/aerogear/graphback/pull/1901#discussion_r472998891

### PR DESCRIPTION
**Pull request by ssd71**
This PR adds global conflict configuration for datasync

fixes #1902

### CODE REVIEW

File: docs/datasync/conflict-resolution-intro.md [hunk 1]
```diff
@@ -88,17 +88,18 @@ const {
   modelDefs,
   {
     db,
-    dataSyncConflictMap: {
-      Comment: {
-        enabled: true,
-        deltaTTL: 604800
+    conflictConfig: {
+      configMap:{
```
Comment (wtrocki) [inline docs/datasync/conflict-resolution-intro.md:92]: Change to something more obvious..Models?

File: docs/datasync/conflict-resolution-intro.md [hunk 2]
```diff
@@ -88,17 +88,18 @@ const {
...
+        Comment: {
+          enabled: true
```
Comment (wtrocki) [inline docs/datasync/conflict-resolution-intro.md:94]: Demo global config

**Review by wtrocki — COMMENTED** [review]
Looks good. need minor changes and later will verify

File: docs/datasync/conflict-resolution-intro.md (hunk 1 above)
Comment (ssd71) [reply docs/datasync/conflict-resolution-intro.md]: Thanks!

File: docs/datasync/conflict-resolution-intro.md (hunk 2 above)
Comment (ssd71) [reply docs/datasync/conflict-resolution-intro.md]: Thanks!

File: packages/graphback-datasync/tests/getModelConfigTest.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      deltaTTL: 172800,
+    });
+
+    expect(getModelConfigFromGlobal('Test', { enabled: true, models: { Test: { deltaTTL: 30, conflictResolution: ServerSideWins }} })).toMatchObject({
+      enabled: true,
+      conflictResolution: ServerSideWins,
+      deltaTTL: 30,
+    });
+  })
+})
```
Comment (machi1990) [inline packages/graphback-datasync/tests/getModelConfigTest.ts:31]: <3 Thank you for adding these!

File: packages/graphql-serve/src/runtime.ts [hunk 4]
```diff
@@ -70,11 +54,14 @@ export const createRuntime = async (modelDir: string, db: Db, datasyncServeConfi
...
-    const modelConflictMap = getConflictMapForModels(model, conflictResolutionStrategy, deltaTTL);
 
     graphbackAPI = createDataSyncAPI(model, {
       db,
-      dataSyncConflictMap: modelConflictMap
+      conflictConfig: {
+        enabled: true,
+        conflictResolution: conflictResolutionStrategy,
+        deltaTTL
+      }
```
Comment (machi1990) [inline packages/graphql-serve/src/runtime.ts:64]: :+1:

File: templates/ts-apollo-mongodb-datasync-backend/src/index.ts [hunk 5]
```diff
@@ -26,7 +26,7 @@ async function start() {
 
   const db = await connectDB()
 
-  const { typeDefs, resolvers, contextCreator } = createDataSyncAPI(modelDefs, { db, dataSyncConflictMap:{ Comment: { enabled: true, deltaTTL: 604800 }}});
+  const { typeDefs, resolvers, contextCreator } = createDataSyncAPI(modelDefs, { db, conflictConfig: { configMap: { Comment: { enabled: true } } } });
```
Comment (machi1990) [inline templates/ts-apollo-mongodb-datasync-backend/src/index.ts:29]:
```suggestion
  const { typeDefs, resolvers, contextCreator } = createDataSyncAPI(modelDefs, { db, conflictConfig: { models: { Comment: { enabled: true } } } });
```
?

Comment (machi1990) [reply templates/ts-apollo-mongodb-datasync-backend/src/index.ts:29]: Should it be `models` instead of `configMap`?

Comment (ssd71) [reply templates/ts-apollo-mongodb-datasync-backend/src/index.ts:29]: Thanks for catching that!

File: docs/datasync/conflict-resolution-intro.md [hunk 6]
```diff
@@ -88,17 +88,14 @@ const {
...
-      }
+    conflictConfig: {
+      enabled: true
     }
   }
 );
 ```
 
-The `dataSyncConflictMap` argument is used to `enable` the server-side Conflict resolution feature for specific models, in addition to Delta Queries. When it is enabled, a `delta` table is maintained for each model which uses it. 
+The `conflictConfig` argument is used to either configure conflicts for all `@datasync` enabled models by directly setting global conflict parameters like `enabled` or configuring specific models via a `models` property. When conflicts are enabled for a specific model, a delta table is maintained for that model.
```
Comment (machi1990) [inline docs/datasync/conflict-resolution-intro.md:98]: I'd also add an example like https://github.com/aerogear/graphback/pull/1905/files#diff-36beda04046dab190232c0cf12a2999eR91 to show how one can provide custom model conflict config.

Comment (machi1990) [reply docs/datasync/conflict-resolution-intro.md:98]: What do you think?

Comment (ssd71) [reply docs/datasync/conflict-resolution-intro.md:98]: I have added an example here: https://github.com/aerogear/graphback/blob/c45fe8b3bc0907930fec23e418b7ed8a005d5d56/docs/datasync/conflict-resolution-strategies.md

Does that look okay?

Comment (machi1990) [reply docs/datasync/conflict-resolution-intro.md:98]: Ah right. That'll do. Maybe a link to that part will do?

Comment (ssd71) [reply docs/datasync/conflict-resolution-intro.md:98]: There's a link in the same file, under [this](https://github.com/aerogear/graphback/blob/c45fe8b3bc0907930fec23e418b7ed8a005d5d56/docs/datasync/conflict-resolution-intro.md#modify-the-template-to-use-createdatasyncapi) heading, near the end, would that be okay?

Comment (machi1990) [reply docs/datasync/conflict-resolution-intro.md:98]: My opinion is that it is kind of hidden by other docs about strategies. But I guess it is okay for now, as I do not have any other better idea. Let's see what other will say or we can revisit later if needed. Thanks!

Comment (wtrocki) [reply docs/datasync/conflict-resolution-intro.md:98]: Yep. This is convoluted. I will refactor docs once we get something out there as they need some simplification aggregation of concepts you do normally and things that you do only in advanced cases.

**Review by wtrocki — APPROVED** [review]

File: docs/datasync/conflict-resolution-intro.md (hunk 6 above)
Comment (machi1990) [reply docs/datasync/conflict-resolution-intro.md:98]: Sounds good. 

Again this is really good work. Thanks @ssd71 @wtrocki

File: templates/ts-apollo-mongodb-datasync-backend/src/index.ts [hunk 7]
```diff
@@ -26,7 +26,7 @@ async function start() {
 
   const db = await connectDB()
 
-  const { typeDefs, resolvers, contextCreator } = createDataSyncAPI(modelDefs, { db, dataSyncConflictMap:{ Comment: { enabled: true, deltaTTL: 604800 }}});
+  const { typeDefs, resolvers, contextCreator } = createDataSyncAPI(modelDefs, { db, conflictConfig: { models: { Comment: { enabled: true } } } });
```
Comment (machi1990) [inline templates/ts-apollo-mongodb-datasync-backend/src/index.ts:29]: Unrelated to this change but I noticed that https://github.com/aerogear/graphback/blob/c45fe8b3bc0907930fec23e418b7ed8a005d5d56/templates/ts-apollo-mongodb-datasync-backend/src/index.ts#L13 is not used anywhere. Can we clean that one up if its not too late?

Comment (ssd71) [reply templates/ts-apollo-mongodb-datasync-backend/src/index.ts:28]: Thanks!

**Review by machi1990 — APPROVED** [review]

**Comment by machi1990** [comment]
Thank you @ssd71!

**Pull request merged**